File: examples/docs-code-snippets/src/main/java/examples/java/application/FireEventAndResolveOnEvent.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;
import com.ing.baker.runtime.javadsl.EventInstance;
import examples.java.events.OrderPlaced;

public class FireEventAndResolveOnEvent {

    private final Baker baker;

    public FireEventAndResolveOnEvent(Baker baker) {
        this.baker = baker;
    }

    public void example(String recipeInstanceId, OrderPlaced orderPlaced) {
        var eventInstance = EventInstance.from(orderPlaced);
        var sensoryEventResult = baker.fireEventAndResolveOnEvent(recipeInstanceId, eventInstance, "ExpectedEvent").join();
    }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/application/FireEventAndResolveOnEvent.kt
package examples.kotlin.application

import com.ing.baker.runtime.javadsl.EventInstance
import com.ing.baker.runtime.kotlindsl.Baker
import examples.kotlin.events.OrderPlaced
import kotlin.time.Duration.Companion.seconds

class FireEventAndResolveOnEvent(private val baker: Baker) {

    suspend fun example(recipeInstanceId: String, orderPlaced: OrderPlaced) {
        val orderPlacedEvent = EventInstance.from(orderPlaced)
        baker.fireSensoryEventAndAwaitReceived(recipeInstanceId, orderPlacedEvent)
        baker.awaitEvent(recipeInstanceId, "ExpectedEvent", 5.seconds)
    }
}